python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -148,6 +148,22 @@ def get_lsst_photometric_uncertainties(self):
                                                  'lsstBandpassDict')
 
 
+    def calculateVisibility(self, magFilter, m5, sigma=0.12):
+        """
+        Calculate the probability of detecting a particular source.
+
+        @ param [in] magFilter is the magnitude of the object in the given filter.
+
+        @ param [in] obs_metadata is the observation metadata for a given visit (to determine m5).
+
+        @ param [out] visibility (0/1).
+        """
+        if len(magFilter) == 0:
+            return numpy.array([])
+        completeness = 1.0 / (1 + numpy.exp((magFilter - m5)/sigma))
+        probability = numpy.random.random_sample(len(magFilter))
```

Comment:
We don't have any instance catalogs that use random seeds, yet.  @rbiswas4 and @jbkalmbach are trying to figure out how to deal with this with supernovae and AGN, respectively.  No solution has been reached.  The best idea we've had is: re-seed numpy.random with objId for each object and just draw one random number from the resulting distribution.  Of course each supernova/agn just needs a few random numbers representing the parameters characterizing the light curve of the object.  Presumably, you need one random number for each permutation of objId + MJD (to account for the randomness involved in viewing one object on multiple occasions).